### PR TITLE
add tls authentication for NATS Streaming

### LINKED ISSUE #2296: NATS Authentication Support
### Proposal

Currently NATS isn't supporting Authentication as per documents https://keda.sh/docs/1.4/scalers/nats-streaming/#authentication-parameters, we would like to see support for TLS Mutual Authentication

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Adarsh-verma-14**
Add tls authentication for NATS Streaming scaler.

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #2296

### CODE REVIEW

File: pkg/scalers/stan_scaler.go [hunk 1]
```diff
@@ -73,12 +82,12 @@ func NewStanScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error parsing stan metadata: %w", err)
 	}
-
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, stanMetadata.enableTLS)
```
Comment (JorTurFer) [inline pkg/scalers/stan_scaler.go:85]: This change disables the CA validation on client side (`InsecureSkipVerify`), is this intended?

Comment (Adarsh-verma-14) [reply pkg/scalers/stan_scaler.go:85]: Hi @JorTurFer,  No, disabling the CA validation on client side is not intended.

Comment (JorTurFer) [reply pkg/scalers/stan_scaler.go:85]:
```suggestion
	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)

	if meta.enableTLS {
		config, err := kedautil.NewTLSConfig(meta.cert, meta.key, meta.ca, false)
		if err != nil {
			return nil, err
		}
		httpClient.Transport = kedautil.CreateHTTPTransportWithTLSConfig(config)
	}
```

File: pkg/scalers/stan_scaler_test.go [hunk 2]
```diff
@@ -60,6 +91,34 @@ func TestStanParseMetadata(t *testing.T) {
...
+				t.Errorf("Expected ca to be set to %v but got %v\n", testData.authParams["ca"], meta.enableTLS)
+			}
+			if meta.cert != testData.authParams["cert"] {
+				t.Errorf("Expected cert to be set to %v but got %v\n", testData.authParams["cert"], meta.cert)
+			}
+			if meta.key != testData.authParams["key"] {
+				t.Errorf("Expected key to be set to %v but got %v\n", testData.authParams["key"], meta.key)
+			}
+
+		}
```
Comment (JorTurFer) [inline pkg/scalers/stan_scaler_test.go:118]:
```suggestion
			}
		}
```

**Comment by JorTurFer** [comment]
LGTM! 
Could you open an PR adding these extra parameters to the docs?

**Comment by Adarsh-verma-14** [comment]
> LGTM! Could you open an PR adding these extra parameters to the docs?

sure

**Comment by Adarsh-verma-14** [comment]
@JorTurFer, I raised the docs PR regarding this could you review please ?
[https://github.com/kedacore/keda-docs/pull/1354](url)

**Comment by Adarsh-verma-14** [comment]
Hi @JorTurFer , I have raised the docs PR regarding these changes. PR link- https://github.com/kedacore/keda-docs/pull/1354

**Review by zroubalik — APPROVED** [review]

**Pull request merged**